### PR TITLE
Remove --boxed and pytest-forked dependency

### LINKED ISSUE #468: deprecate and remove the --boxed option
the related code has been moved to the pytest-forked plugin and wont ever support coverage or windows sanely

### LINKED ISSUE #543: drop --boxed and deprecate looponfail
As the 2.x release is incoming we should take care of that

Looponfail should eventually be reimplemented in core using subprocess and a hook to register the file system watchers (so pytest plugins could support Realtime Filesystem watching)

### PR DESCRIPTION
**Pull request by nicoddemus**
Fixes #468
Fixes #543

### CODE REVIEW

**Review by asottile — APPROVED** [review]

**Comment by nicoddemus** [comment]
Not so simple I'm afraid... pytest-xdist's own code still uses some `py` functionality, such as `py.log`, `py.local`, etc.

Will go ahead and try to remove that dependency.

**Comment by asottile** [comment]
looks like the `py.log` can be replaced pretty easily with just `functools.partial(print, file=sys.stderr)` (or `lambda *a, **k: None` in the non-debug case) --  I'll send a patch for that one

**Comment by nicoddemus** [comment]
Yeah I was working on that.

Unfortunately just hit a wall: `pytest_xdist_make_scheduler` receives a `py.log` instance. I guess we will need a round of deprecation for that parameter. 😬

**Comment by asottile** [comment]
> Yeah I was working on that.
> 
> Unfortunately just hit a wall: `pytest_xdist_make_scheduler` receives a `py.log` instance. I guess we will need a round of deprecation for that parameter. 😬

I made a class which looks the same as the `py.log` thing -- and kept the usages the same

**Comment by nicoddemus** [comment]
Cool.

I have some WIP replacing `py.local` with `Path` too, but I won't be able to push it in the next few hours... feel free to open a separate PR replacing `py.log` though.

**Comment by asottile** [comment]
alrighty #822 and #823 should be good to go now I think?

**Review by asottile — APPROVED** [review]

**Review by asottile — APPROVED** [review]

File: setup.cfg [hunk 1]
```diff
@@ -44,7 +44,7 @@ python_requires = >=3.6
 install_requires =
     execnet>=1.1
     pytest>=6.2.0
-    pytest-forked
+    py
```
Comment (nicoddemus) [inline setup.cfg:47]: We still use `py` in pytest-xdist's own code, which is being fixed by #822, #823, and #824.

File: changelog/468.trivial.rst [hunk 2]
```diff
@@ -0,0 +1 @@
+The ``py`` dependency has been dropped.
```
Comment (nicoddemus) [inline changelog/468.trivial.rst:1]: AFAICT we no longer depend on `py`. 😁

**Comment by nicoddemus** [comment]
Rebased and ready for final review.

**Review by RonnyPfannschmidt — APPROVED** [review]
💥

**Pull request merged**